src/components/HeatMap/index.tsx

```diff
@@ -0,0 +1,282 @@
+
+    const [isLoading, setIsLoading] = useState(false);
+
+    useEffect(() => {
+        setIsLoading(true);
+        if (city === "df") {
+            setCrimes();
+            setIsLoading(false);
+        } else {
+            setCrimes();
+            getSpCoordinates().then((res) => setIsLoading(false));
+        }
+    }, []);
+
+    const getSpCoordinates = async () => {
+        const response = await getCoordinates(city);
+
+        setCoordinatesSp(
+            response.body[0].cities.map((cityCoord: CityCoordinate) => {
+                if (cityCoord.name !== "Cândido Mota")
+                    cityCoord.coordinates = cityCoord.coordinates.map(
+                        (coordinateCity) => {
+                            return {
+                                longitude: Number(coordinateCity[0]),
+                                latitude: Number(coordinateCity[1]),
+                            };
+                        }
+                    );
+                return {
+                    name: cityCoord.name,
+                    coordinates: cityCoord.coordinates,
+                };
+            })
+        );
+    };
+
+    const setCrimes = () => {
+        if (secretaryOccurrences) {
+            setCitiesCrimes(
+                secretaryOccurrences.map((city) => {
+                    const crime = {
+                        name: city.name,
+                        nature: city.crimes[0].nature,
+                        quantity: city.crimes[0].quantity,
+                        classification: city.crimes[0].classification,
+                        color: getHeatColor(city.crimes[0].classification),
+                    };
+
+                    return crime;
+                })
+            );
+        }
+    };
+
+    const getHeatColor = (classification: number) => {
+        let color = "";
+        if (classification === 1) color = theme.heatMapGreen1;
+        else if (classification === 2) color = theme.heatMapYellow2;
+        else if (classification === 3) color = theme.heatMapOrange3;
+        else if (classification === 4) color = theme.heatMapOrange4;
+        else if (classification === 5) color = theme.heatMapOrange5;
+        else if (classification === 6) color = theme.heatMapRed6;
+
+        return color;
+    };
+
+    return (
+        <>
+            {isLoading ? (
+                <CircularLoader size={40} />
+            ) : (
+                <StayNormalMap
+                    loadingEnabled
+                    initialRegion={{
+                        latitude: city === "df" ? -15.780311 : -23.6821604,
+                        longitude: city === "df" ? -47.768043 : -46.8754825,
+                        latitudeDelta: 3,
+                        longitudeDelta: 3,
+                    }}
+                    customMapStyle={
+                        theme.type === "dark" ? staySafeDarkMapStyle : []
+                    }
+                >
+                    {city === "df"
```

Comment:
This conditional is really necessary? Could you make somewhat generic? Cause it has a lot of duplicate code.